### DISCUSSION TITLE
Spaces shortcuts

### DISCUSSION THREAD

**Discussion by bvilnis** [rxhanson/Rectangle#350] [Ideas]
Firstly, love the app. So much better than the paid solution I was previously using.

Any chance you have thought about adding shortcuts to move windows between spaces? I'm a "fullscreen windows across several spaces" kind of guy. The only thing missing in Rectangle's arsenal for me is the ability to move my windows left and right between spaces.

Thanks for the amazing work!

**Comment by rxhanson** [comment]
Thanks for the kind words! This is a popular request.

Apple doesn't have a public API available for Spaces. There's only a private API. Amethyst and SizeUp both use the private API to provide this feature, and last time I tried it in SizeUp (a long time ago) it didn't work for me. The choice not to use private APIs in Rectangle is the choice of more stability and less time spent on quirky support issues. It's also a decent amount of code to add in (and maintain).

With all of that said, I would consider an elegant pull request for it and make it known that it might just not work for some users.

**Comment by bvilnis** [comment]
Ahh ok. I'm not familiar with Apple architecture and APIs. The APIs being private is a pain and I can see why the feature is not currently worth the overhead brain damage.

Thanks for the reply!
